File: solutions/scala/src/main/scala/TimeUtility.scala
import java.time.LocalTime

object TimeUtility {
  def timeOfDay(clock: Clock) : String = {
    clock.time() match {
      case night if night.getHour >= 0 && night.getHour < 6 => "Night"
      case morning if morning.getHour >= 6 && morning.getHour < 12 => "Morning"
      case afternoon if afternoon.getHour >= 12 && afternoon.getHour < 18 => "Afternoon"
      case _ => "Evening"
    }
  }
}

trait Clock {
  def time(): LocalTime = LocalTime.now()
}

File: solutions/php/src/TimeUtility.php
<?php

class TimeUtility {

  public static function timeOfDay($clock) {

    $hour = intval($clock->time()->format("H"));

    if($hour >= 0 && $hour < 6) {
      return "Night";
    }

    if($hour >= 6 && $hour < 12) {
      return "Morning";
    }

    if($hour >= 12 && $hour < 18) {
      return "Afternoon";
    }

    return "Evening";

  }
 
}

abstract class Clock {
  public function time() {
    return new DateTime();
  }
}

?>